### PR TITLE
Loki: query splitting: handle stats

### PR DESCRIPTION
**Pull request by gabor**
when combining multiple chunks of dataframes in the loki-query-split scenario, we need to handle the `stats` arriving for every chunk.

these stats are shown at two places:
1. in the logs panel we show the `Total bytes processed:` at the top
2. in the query-inspector's `Stats` section.

this PR correctly updates the `Total bytes processed` value, and removes every other value.

the reason is, there's no generic way to "merge" stats (for example, how do you merge `bytes processed per second` values), so for now we make sure the most important value is correct, and we can add more values, if it becomes necessary.

(i added a note about possible long-term steps about this to https://github.com/grafana/grafana/issues/63174)

how to test:
1. make a logs request using loki so that chunking will happen
2. verify that the value shown at the top of the logs-panel keeps changing
3. verify that the query-inspector only shows the single total-bytes-processed value in the stats section.

### CODE REVIEW

**Review by matyax — COMMENTED** [review]
Looks good to me! I think we should update the tests after this.

File: public/app/plugins/datasource/loki/queryUtils.ts [hunk 1]
```diff
@@ -345,28 +345,33 @@ function combineFrames(dest: DataFrame, source: DataFrame) {
...
-  if (!source.meta?.stats) {
-    return;
+const TOTAL_BYTES_STAT = 'Summary: total bytes processed';
+
+function getCombinedMetadataStats(
+  destStats: QueryResultMetaStat[],
+  sourceStats: QueryResultMetaStat[]
+): QueryResultMetaStat[] {
+  // in the current approach, we only handle a single stat
+  const destStat = destStats.find((s) => (s.displayName = TOTAL_BYTES_STAT));
```
Comment (leeoniya) [inline public/app/plugins/datasource/loki/queryUtils.ts:361]: should be `===` ?

File: public/app/plugins/datasource/loki/queryUtils.ts [hunk 2]
```diff
@@ -345,28 +345,33 @@ function combineFrames(dest: DataFrame, source: DataFrame) {
...
+  const sourceStat = sourceStats.find((s) => (s.displayName = TOTAL_BYTES_STAT));
```
[the remark this answers is not in the document]
Comment (gabor) [reply public/app/plugins/datasource/loki/queryUtils.ts]: yeah, good catch! i cannot believe i did not get a warning about that one.

File: public/app/plugins/datasource/loki/queryUtils.ts (hunk 1 above)
Comment (gabor) [reply public/app/plugins/datasource/loki/queryUtils.ts]: yup, thanks , fixed!

File: public/app/plugins/datasource/loki/queryUtils.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (gabor) [reply public/app/plugins/datasource/loki/queryUtils.ts]: fixed

**Comment by gabor** [comment]
update the existing tests and added more tests.

**Review by matyax — APPROVED** [review]

**Pull request merged**